src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.cs

```diff
@@ -159,7 +160,14 @@ protected override SyntaxToken GetMethodNameAtInvocation(IEnumerable<SyntaxNodeO
             if (!leadingTrivia.Any(t => t.IsKind(SyntaxKind.EndOfLineTrivia)) && !methodDefinition.FindTokenOnLeftOfPosition(methodDefinition.SpanStart).IsKind(SyntaxKind.OpenBraceToken))
             {
                 var originalMethodDefinition = methodDefinition;
-                methodDefinition = methodDefinition.WithPrependedLeadingTrivia(SpecializedCollections.SingletonEnumerable(SyntaxFactory.CarriageReturnLineFeed));
+                var options = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+                methodDefinition = methodDefinition.WithPrependedLeadingTrivia(SpecializedCollections.SingletonEnumerable(SyntaxFactory.EndOfLine(options.GetOption(FormattingOptions2.NewLine))));
+
+                if (!originalMethodDefinition.FindTokenOnLeftOfPosition(originalMethodDefinition.SpanStart).TrailingTrivia.Any(SyntaxKind.EndOfLineTrivia))
+                {
+                    // Add a second new line since there were no line endings in the original form
+                    methodDefinition = methodDefinition.WithPrependedLeadingTrivia(SpecializedCollections.SingletonEnumerable(SyntaxFactory.EndOfLine(options.GetOption(FormattingOptions2.NewLine))));
+                }
```

Comment:
why did this logic need to change?

Reply:
Previously the formatter ran before this call, and inserted one of the two newlines.

Reply:
Example: the formatter would not let this exist:

```
NewMethod();void NewMethod()
{
}
```

and would add the first new line to give this:

```
NewMethod();
void NewMethod()
{
}
```